homeassistant/components/smartthings/climate.py

```diff
@@ -499,12 +500,20 @@ def temperature_unit(self) -> str:
 
     def _determine_swing_modes(self) -> list[str]:
         """Return the list of available swing modes."""
+        supported_swings = [SWING_OFF]
         supported_modes = self._device.status.attributes[
             Attribute.supported_fan_oscillation_modes
         ][0]
-        supported_swings = [
-            FAN_OSCILLATION_TO_SWING.get(m, SWING_OFF) for m in supported_modes
-        ]
+        if supported_modes is not None:
+            supported_swings = [
+                FAN_OSCILLATION_TO_SWING.get(m, SWING_OFF) for m in supported_modes
+            ]
+        # this is a workaround as Samsung sometimes return none for supportedFanOscillationModes
+        elif (
+            self._device.status.attributes[Attribute.fan_oscillation_mode][0]
+            is not None
+        ):
+            supported_swings = [SWING_OFF, SWING_BOTH, SWING_VERTICAL, SWING_HORIZONTAL]
```

Comment:
This would just add all available options if it finds a current swing mode?
That seems wrong?

Reply:
As supportedFanOscillationModes is None even though the AC support multiple swing modes, there is no way to determine what mode this AC support, so I added all the possible modes.

Reply:
No problem, I'll fix this, and I'll look at the upstream library to see if there is anything to do, and open a new merge request for that